main.c

```diff
@@ -811,7 +811,11 @@ int __attribute__((used)) main(void) {
     // Create a new filesystem only if we're not in a safe mode.
     // A power brownout here could make it appear as if there's
     // no SPI flash filesystem, and we might erase the existing one.
-    filesystem_init(safe_mode == NO_SAFE_MODE, false);
+
+    // Check whether CIRCUITPY is available. Don't check if it already hasn't been found.
+    if (safe_mode != NO_CIRCUITPY && !filesystem_init(safe_mode == NO_SAFE_MODE, false)) {
```

Comment:
im one of those people who likes parens around each clause 

`if ((safe_mode != NO_CIRCUITPY) && !filesystem_init(safe_mode == NO_SAFE_MODE, false)) {`